hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/file/FileTableFunction.java

```diff
@@ -83,6 +95,52 @@ private static String randomName() {
         return SCHEMA_NAME_FILES + "_" + UuidUtil.newUnsecureUuidString().replace('-', '_');
     }
 
+    @Override
+    public List<Permission> permissions(SqlCall call) {
+        // Try find operand by name first, most common option.
+        SqlNode astPath = findOperandByName(OPTION_PATH, call);
+        // If not successful -> try to find by position: SELECT * FROM TABLE(fn('path', 'glob'))
+        if (astPath == null) {
+            astPath = findOperandByPosition(0, call);
+        }
+        if (astPath == null) {
+            throw QueryException.error("Can't determine the file path, query is denied in secure environment");
+        }
```

Comment:
When can this happen? Do you have a sample query for this?

Reply:
Yes : https://github.com/hazelcast/hazelcast-enterprise/pull/6695/files#diff-db1fe8bc37f27fa36d546e307a2c0fa1512c69306915123c26f2b85f0840e0e8R157